iceoryx2/tests/service_tests.rs

```diff
@@ -53,6 +63,230 @@ mod service {
         assert_that!(received_event[0], eq EVENT_ID);
     }
 
+    trait SutFactory {
+        type Factory;
+        type CreateError: std::fmt::Debug;
+        type OpenError: std::fmt::Debug;
+
+        fn create(service_name: &ServiceName) -> Result<Self::Factory, Self::CreateError>;
+        fn open(service_name: &ServiceName) -> Result<Self::Factory, Self::OpenError>;
+    }
+
+    impl<Sut: Service> SutFactory for publish_subscribe::PortFactory<Sut, u64> {
+        type Factory = publish_subscribe::PortFactory<Sut, u64>;
+        type CreateError = PublishSubscribeCreateError;
+        type OpenError = PublishSubscribeOpenError;
+
+        fn create(service_name: &ServiceName) -> Result<Self::Factory, Self::CreateError> {
+            Sut::new(&service_name).publish_subscribe().create::<u64>()
+        }
+
+        fn open(service_name: &ServiceName) -> Result<Self::Factory, Self::OpenError> {
+            Sut::new(&service_name).publish_subscribe().open::<u64>()
+        }
+    }
+
+    impl<Sut: Service> SutFactory for event::PortFactory<Sut> {
+        type Factory = event::PortFactory<Sut>;
+        type CreateError = EventCreateError;
+        type OpenError = EventOpenError;
+
+        fn create(service_name: &ServiceName) -> Result<Self::Factory, Self::CreateError> {
+            Sut::new(&service_name).event().create()
+        }
+
+        fn open(service_name: &ServiceName) -> Result<Self::Factory, Self::OpenError> {
+            Sut::new(&service_name).event().open()
+        }
+    }
+
+    fn concurent_creating_services_with_unique_names_is_successful_impl<Factory: SutFactory>() {
+        let _watch_dog = Watchdog::new_with_timeout(Duration::from_secs(30));
+        const NUMBER_OF_THREADS: usize = 4;
+        const NUMBER_OF_ITERATIONS: usize = 50;
+
+        let barrier_enter = Barrier::new(NUMBER_OF_THREADS);
+        let barrier_exit = Barrier::new(NUMBER_OF_THREADS);
+
+        std::thread::scope(|s| {
+            let mut threads = vec![];
+            for _ in 0..NUMBER_OF_THREADS {
+                threads.push(s.spawn(|| {
+                    for _ in 0..NUMBER_OF_ITERATIONS {
+                        barrier_enter.wait();
+
+                        let service_name = generate_name();
+                        let _sut = Factory::create(&service_name).unwrap();
+
+                        barrier_exit.wait();
+                    }
+                }));
+            }
+
+            for thread in threads {
+                thread.join().unwrap();
+            }
+        });
+    }
+
+    #[test]
+    fn concurrent_creating_services_with_unique_names_is_successful<Sut: Service>() {
+        concurent_creating_services_with_unique_names_is_successful_impl::<
```

Comment:
Could you please split those up in two tests. One for event and one for publish_subscribe?

Reply:
I realized that sometimes one variation fails and the other doesn't so when the CI fails you can - when its need to be - at least try to fix it without running a native target.